src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue

```diff
@@ -1,26 +1,147 @@
+        <p-data-loader class="chart-wrapper"
+                       :loading="loading"
+        >
+            <div ref="chartRef"
+                 class="chart"
+            />
+        </p-data-loader>
+    </div>
 </template>
 
 <script lang="ts">
 import {
-    defineComponent, reactive, toRefs,
+    computed,
+    defineComponent, onUnmounted, reactive, toRefs, watchEffect,
 } from 'vue';
 
+import * as am5 from '@amcharts/amcharts5';
+import type { Root } from '@amcharts/amcharts5';
+import type { IPieSeriesSettings } from '@amcharts/amcharts5/percent';
+import type { IXYSeriesSettings } from '@amcharts/amcharts5/xy';
+import { PDataLoader } from '@spaceone/design-system';
+import { random, range } from 'lodash';
+
+import { useAmcharts5 } from '@/common/composables/amcharts5';
+import type { ChartType } from '@/common/composables/amcharts5/type';
+import { CHART_TYPE, DATE_VALUE_FIELD } from '@/common/composables/amcharts5/type';
+
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface Props {
 }
 
+// TODO: sample data
+const SAMPLE_XY_CHART_DATA = range(4).map((d) => ({
+    date: `2022-${11 + d}`,
+    projectA: random(10000, 50000),
+    projectB: random(10000, 50000),
+    projectC: random(10000, 50000),
+}));
+const SAMPLE_PIE_CHART_DATA = [
+    { category: 'google cloud', value: random(1000, 5000) },
+    { category: 'aws', value: random(1000, 5000) },
+    { category: 'azure', value: random(1000, 5000) },
+    { category: 'alibaba', value: random(1000, 5000) },
+];
+
 export default defineComponent<Props>({
     name: 'BaseTrendWidget',
-    components: {},
-    props: {},
-    setup() {
-        const state = reactive({});
+    components: {
+        PDataLoader,
+    },
+    props: {
+        chartType: { // TODO: just for example
+            type: String,
+            default: 'LINE',
+        },
+    },
+    setup(props) {
+        const state = reactive({
+            loading: false,
+            chartRef: null as HTMLElement | null,
+            chartRoot: computed(() => root.value),
+            chartData: SAMPLE_XY_CHART_DATA, // TODO: mock data
+            timeUnit: 'month', // TODO: mock data
+            labels: ['projectA', 'projectB', 'projectC'], // TODO: mock data
+            chartSeriesDataProcessor: computed<undefined | am5.DataProcessor>(() => {
+                if (!state.chartRoot) return undefined;
+                return am5.DataProcessor.new(state.chartRoot, {
+                    dateFormat: 'yyyy-M', // TODO: will be changed dynamically
+                    dateFields: [DATE_VALUE_FIELD],
+                });
+            }),
+        });
+
+        const {
+            root,
+            createXYDateChart,
+            createPieChart,
+            createDonutChart,
+            createPieSeries,
+            createXYLineSeries,
+            createXYStackedColumnSeries,
+            disposeRoot,
+        } = useAmcharts5(computed(() => state.chartRef));
```

Comment:
```suggestion
        } = useAmcharts5(toRef(state, 'chartRef'));
```

이런 기능을 통해 reactive 객체의 한 개 프로퍼티를 ref 로 변경 가능합니다.